src/App.tsx

```diff
@@ -15,36 +15,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+  None = 'none',
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState<boolean>(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+
+  const reverseGoods = () => {
+    setReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setSortType(SortType.None);
+    setReversed(false);
+  };
+
+  const isResetVisible = sortType !== SortType.None || isReversed;
+
+  const newGoods = useMemo(() => {
```

Comment:
Create `getSortedGoods` function and call it here
Move this function outside the component